build.go

```diff
@@ -218,11 +218,18 @@ func main() {
 		os.Setenv("GOPATH", gopath)
 		log.Println("GOPATH is", gopath)
 	} else {
-		gopath, _ = filepath.EvalSymlinks(gopath)
+		inside := false
 		wd, _ := os.Getwd()
 		wd, _ = filepath.EvalSymlinks(wd)
-		if filepath.Join(gopath, "src/github.com/syncthing/syncthing") != wd {
-			fmt.Println("You are running this outside of GOPATH/src/github.com/syncthing/syncthing, this might cause failure!")
+		for _, p := range filepath.SplitList(gopath) {
+			p, _ = filepath.EvalSymlinks(p)
+			if filepath.Join(p, "src/github.com/syncthing/syncthing") == wd {
+				inside = true
+				break
+			}
+		}
+		if !inside {
+			fmt.Println("You seem to have GOPATH set but the Syncthing source not placed correctly within it, which may cause problems.")
```

Comment:
I didn't like your "proceed with caution" part that well, as the user can't really do anything anymore at that point, cautiously or not (except for hitting ctrl-c in a hurry). I hope I didn't just make it worse.

Reply:
It looks perfectly fine to me :)